src/code/z_actor.c

```diff
@@ -456,232 +469,236 @@ void Target_SetColors(TargetContext* targetCtx, Actor* actor, s32 type, PlayStat
-            projectedPos.y = CLAMP(projectedPos.y, -SCREEN_HEIGHT, SCREEN_HEIGHT);
+        projectedPos.y = ((SCREEN_HEIGHT / 2) * (projectedPos.y * invW)) * var1;
+        projectedPos.y = CLAMP(projectedPos.y, -SCREEN_HEIGHT, SCREEN_HEIGHT);
 
-            projectedPos.z = projectedPos.z * var1;
+        projectedPos.z = projectedPos.z * var1;
 
-            targetCtx->unk4C--;
-            if (targetCtx->unk4C < 0) {
-                targetCtx->unk4C = 2;
-            }
+        targetCtx->currentLockOnIndex--;
+        if (targetCtx->currentLockOnIndex < 0) {
+            targetCtx->currentLockOnIndex = ARRAY_COUNT(targetCtx->lockOnEntries) - 1;
+        }
 
-            Target_SetPos(targetCtx, targetCtx->unk4C, projectedPos.x, projectedPos.y, projectedPos.z);
+        Target_SetLockOnPos(targetCtx, targetCtx->currentLockOnIndex, projectedPos.x, projectedPos.y, projectedPos.z);
 
-            if ((!(player->stateFlags1 & PLAYER_STATE1_40)) || (actor != player->targetedActor)) {
-                OVERLAY_DISP = Gfx_SetupDL(OVERLAY_DISP, SETUPDL_57);
+        if (!(player->stateFlags1 & PLAYER_STATE1_40) || (actor != player->targetedActor)) {
+            OVERLAY_DISP = Gfx_SetupDL(OVERLAY_DISP, SETUPDL_57);
 
-                for (spB0 = 0, spAC = targetCtx->unk4C; spB0 < spB8; spB0++, spAC = (spAC + 1) % 3) {
-                    entry = &targetCtx->unk50[spAC];
+            for (i = 0, index = targetCtx->currentLockOnIndex; i < totalEntries;
+                 i++, index = (index + 1) % ARRAY_COUNT(targetCtx->lockOnEntries)) {
+                entry = &targetCtx->lockOnEntries[index];
 
-                    if (entry->unkC < 500.0f) {
-                        if (entry->unkC <= 120.0f) {
-                            var2 = 0.15f;
-                        } else {
-                            var2 = ((entry->unkC - 120.0f) * 0.001f) + 0.15f;
-                        }
+                if (entry->distance < 500.0f) {
+                    s32 lockOnIndex;
+
+                    if (entry->distance <= 120.0f) {
+                        var2 = 0.15f;
+                    } else {
+                        var2 = ((entry->distance - 120.0f) * 0.001f) + 0.15f;
+                    }
 
-                        Matrix_Translate(entry->pos.x, entry->pos.y, 0.0f, MTXMODE_NEW);
-                        Matrix_Scale(var2, 0.15f, 1.0f, MTXMODE_APPLY);
+                    Matrix_Translate(entry->pos.x, entry->pos.y, 0.0f, MTXMODE_NEW);
+                    Matrix_Scale(var2, 0.15f, 1.0f, MTXMODE_APPLY);
 
-                        gDPSetPrimColor(OVERLAY_DISP++, 0, 0, entry->color.r, entry->color.g, entry->color.b,
-                                        (u8)alpha);
+                    gDPSetPrimColor(OVERLAY_DISP++, 0, 0, entry->color.r, entry->color.g, entry->color.b, (u8)alpha);
 
-                        Matrix_RotateZS((targetCtx->unk4B * 512), MTXMODE_APPLY);
+                    Matrix_RotateZS(targetCtx->rotation * 512, MTXMODE_APPLY);
```

Comment:
```suggestion
                    Matrix_RotateZS(targetCtx->rotZ * 0x200, MTXMODE_APPLY);
```
Maybe `rotZ`? But it's not true rotation since 1 unit of `rotation` is 0x200 angle units so idk (maybe `rotZIndex`?)

Also worth noting below is the `| 0x80` when applied to `rotation`, but then `0x80 * 0x200` is `0x10000` or 360 degrees, so that connection might be worth noting

Reply:
ohhh, maybe with the `0x80 * 0x200 == 0x10000` stuff is how this extra `0x80` in the rotation is not noticeable at all.
I'll go with `rotZIndex`

Reply:
is it an index though? to me it just seems like it's a binary angle still

Reply:
It definitely is still an angle, just in different units than a binary in a sense. It's like:
`#define CONVEYOR_DIRECTION_TO_BINANG(conveyorDirection) ((conveyorDirection) * (0x10000 / 64))`
Where `conveyorDirection` is an angle ranging from `0` to `0x40`, but this is an angle that ranges from `0` to `0x80`.

Maybe `rotZDirection` or `rotZTick`? `rotZ` alone is usually for binang but maybe it's fine